src/libcrun/utils.c

```diff
@@ -429,8 +430,7 @@ safe_readlinkat (int dfd, const char *name, char **buffer, ssize_t hint, libcrun
       if (tmp_buf != NULL)
         buf_size += 256;
 
-      /* Allocate an extra byte so the buffer can be NUL terminated.  */
-      tmp_buf = xrealloc (tmp_buf, buf_size + 1);
```

Comment:
not sure this is correct.  We can read the whole buffer here, and the set at line 441 can overwrite the last byte.

Reply:
If we read the whole buffer, there will be another allocation because of 

```
 } while (size == buf_size);
```